library/src/scala/quoted/runtime/QuoteMatching.scala

```diff
@@ -40,5 +42,12 @@ trait QuoteMatching:
      *  @param pattern `Type[?]` containing the pattern tree
      *  @return None if it did not match, `Some(tup)` if it matched where `tup` contains `Type[Ti]``
      */
-    def unapply[TypeBindings <: Tuple, Tup <: Tuple](scrutinee: Type[?])(using pattern: Type[?]): Option[Tup]
+    // FIXME: is this change TASTy binary compatible? Or we need to create a new version of the interface?
+    def unapply[TypeBindings, Tup <: Tuple](scrutinee: Type[?])(using pattern: Type[?]): Option[Tup]
```

Comment:
@sjrd @smarter Is this something that could potentially be TASTy compatible. It seems that all calls to this method generated by an older compiler should be unpicklable by the new one.

Reply:
If there's no way it could have been *overridden* in a subclass, then it is indeed a backward compatible change. You can enlarge the bounds of a type parameter.

Reply:
This method is implemented in `QuotesImpl` in the compiler. I guess this means that it is not a TASTy compatible change.